Failing multi-cluster metric test

I get an error when running the `tests/test_metrics.py::test_cluster_metric()` test on Python 3.8.13:

```console
_________________________________________________ test_cluster_metric _________________________________________________

evaluator = <synthgauge.evaluate.Evaluator object at 0x000002F4544ACB80>

    def test_cluster_metric(evaluator):
        """Check clustering metric."""
        evaluator.add_metric('multi_clustered_MSD', random_state=24)
        evaluator.evaluate()
>       assert evaluator.metric_results == pytest.approx({
            'multi_clustered_MSD': 0.00291989251686431
        })
E       AssertionError: assert {'multi_clust...5805624295994} == approx({'mult...31 ± 2.9e-09})
E         comparison failed. Mismatched elements: 1 / 1:
E         Max absolute difference: 0.0002763119544347105
E         Max relative difference: 0.09463086495096182
E         Index               | Obtained              | Expected
E         multi_clustered_MSD | 0.0026435805624295994 | 0.00291989251686431 ± 2.9e-09

tests\test_metrics.py:83: AssertionError
```

It's off by about 0.0003 which isn't the end of the world.

@TomJamesW doesn't see this for a Python 3.6 install but can with Python 3.8. We should check the observed values are the same between our environments.

I get consistent results (0.00264...) which suggests that there isn't some PRNG bleeding and probably has to do with some internal changes to another package. One of:

- `pytest`: unlikely but should check changes to `pytest.approx()` parameter defaults
- `scikit-learn`: this is where the heavy lifting is done for clustering and data generation

These packages are also the only ones with major version differences between Tom's 3.6 env and my 3.8 env.

---

My current `conda` environment is set up as follows:

```yml
name: sg-dev
channels:
  <the usual artifactory channels>
dependencies:
  - asttokens=2.0.5
  - backcall=0.2.0
  - bzip2=1.0.8
  - ca-certificates=2022.5.18.1
  - colorama=0.4.4
  - debugpy=1.6.0
  - decorator=5.1.1
  - executing=0.8.3
  - ipykernel=6.13.1
  - ipython=8.4.0
  - jedi=0.18.1
  - jupyter_client=6.1.12
  - jupyter_core=4.10.0
  - libffi=3.4.2
  - libzlib=1.2.12
  - matplotlib-inline=0.1.3
  - nest-asyncio=1.5.5
  - openssl=3.0.3
  - packaging=21.3
  - parso=0.8.3
  - pickleshare=0.7.5
  - pip=22.1.2
  - prompt-toolkit=3.0.29
  - psutil=5.9.1
  - pure_eval=0.2.2
  - pygments=2.12.0
  - pyparsing=3.0.9
  - python=3.8.13
  - python-dateutil=2.8.2
  - python_abi=3.8
  - pywin32=303
  - pyzmq=22.3.0
  - setuptools=62.3.2
  - six=1.16.0
  - sqlite=3.38.5
  - stack_data=0.2.0
  - tk=8.6.12
  - tornado=6.1
  - traitlets=5.2.2.post1
  - ucrt=10.0.20348.0
  - vc=14.2
  - vs2015_runtime=14.29.30037
  - wcwidth=0.2.5
  - wheel=0.37.1
  - xz=5.2.5
  - pip:
    - alabaster==0.7.12
    - astroid==2.11.5
    - atomicwrites==1.4.0
    - attrs==21.4.0
    - babel==2.10.1
    - certifi==2022.5.18.1
    - cfgv==3.3.1
    - charset-normalizer==2.0.12
    - coverage==6.4.1
    - cycler==0.11.0
    - distlib==0.3.4
    - docutils==0.17.1
    - filelock==3.7.1
    - flake8==4.0.1
    - fonttools==4.33.3
    - identify==2.5.1
    - idna==3.3
    - imagesize==1.3.0
    - iniconfig==1.1.1
    - jinja2==3.1.2
    - joblib==1.1.0
    - kiwisolver==1.4.2
    - kmodes==0.12.1
    - lazy-object-proxy==1.7.1
    - markupsafe==2.1.1
    - matplotlib==3.5.2
    - mccabe==0.6.1
    - nodeenv==1.6.0
    - numpy==1.22.4
    - pandas==1.4.2
    - pillow==9.1.1
    - platformdirs==2.5.2
    - pluggy==1.0.0
    - pre-commit==2.19.0
    - py==1.11.0
    - pycodestyle==2.8.0
    - pyflakes==2.4.0
    - pyqt5==5.15.6
    - pyqt5-qt5==5.15.2
    - pyqt5-sip==12.10.1
    - pytest==7.1.2
    - pytest-cov==3.0.0
    - pytz==2022.1
    - pyyaml==6.0
    - requests==2.27.1
    - scikit-learn==1.1.1
    - scipy==1.8.1
    - seaborn==0.11.0
    - snowballstemmer==2.2.0
    - sphinx==4.3.2
    - sphinx-autoapi==1.8.4
    - sphinx-rtd-theme==1.0.0
    - sphinxcontrib-applehelp==1.0.2
    - sphinxcontrib-devhelp==1.0.2
    - sphinxcontrib-htmlhelp==2.0.0
    - sphinxcontrib-jsmath==1.0.1
    - sphinxcontrib-qthelp==1.0.3
    - sphinxcontrib-serializinghtml==1.1.5
    - threadpoolctl==3.1.0
    - toml==0.10.2
    - tomli==2.0.1
    - typing-extensions==4.2.0
    - unidecode==1.3.4
    - urllib3==1.26.9
    - virtualenv==20.14.1
    - wrapt==1.14.1
```

---

For future reference, I got this output into an environment file with the following command in PowerShell:

```powershell
PS > conda env export -n sg-dev --no-builds | findstr -v "prefix" > environment.yml
```

--- a/src/synthgauge/metrics/cluster.py
+++ b/src/synthgauge/metrics/cluster.py
@@ -53,8 +53,10 @@ def clustered_MSD(combined, synthetic_indicator, method, k, random_state=None):
     """
     if method == 'kmeans':
         combined_numeric = combined.select_dtypes(include='number')
-        clusters = KMeans(n_clusters=k, random_state=random_state).fit(
-            combined_numeric).labels_
+        clusters = KMeans(
+            n_clusters=k, algorithm="elkan", random_state=random_state).fit(
+            combined_numeric
+        ).labels_
     if method == 'kprototype':
         cat_cols = combined.select_dtypes(
             include=['object', 'category']).columns

